### ISSUE TITLE
Use logits per default when explaining models like tabpfn.

### ISSUE THREAD

**Issue by mmschlk** [mmschlk/shapiq#426]
### Describe the workflow you want to enable

If a model has a `predict_logits` function then this should be explained per default.

Explanations for classifiers should happen in the logits space and not in the probabilities space per default. Currently users can pass a model's `predict_logits` function to shapiq and get this explained. However, novice users may pass the model, where we currently are not using the `predict_logits` function as default. 

### Describe your proposed solution

Suggested implementation:

#### Implementation
Prioritize the `predict_logits` function when parsing a model over predict and predict_proba

#### Test
Test weather `predict_logits` gets picked over `predict_proba` and explained for a (lightweight) model that implements both.

### Additional context

This got recently discussed/addressed in tabpfn (https://github.com/PriorLabs/TabPFN/issues/374). Newer tabpfn versions should expose predict_logits now.

### Impact

High (Major improvement)
